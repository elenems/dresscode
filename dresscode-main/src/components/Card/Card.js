import React from "react";
import { Link } from "react-router-dom";
import Icon from "../../utils/Icon";
import PropTypes from "prop-types";
function Card(props) {
  return (
    <div className="card">
      <Link to={props.link}>
        <div className="card__card-content">
          <Icon name={props.name} />
          <div>
            <p>{props.text}</p>
            <span className="card__link">{props.linkText}</span>
          </div>
        </div>
      </Link>
    </div>
  );
}

Card.propTypes = {
  link: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  text: PropTypes.string,
  linkText: PropTypes.string
};

export default React.memo(Card);
