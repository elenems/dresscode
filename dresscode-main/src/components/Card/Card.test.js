import { shallow } from "enzyme";
import React from "react";
import Card from "./Card";

describe("Card", () => {
  const props = {
    link: "link",
    name: "name"
  };
  it("Matches snapshot", () => {
    const wrapper = shallow(<Card {...props} />).debug();
    expect(wrapper).toMatchSnapshot();
  });

  it("Has container with class card", () => {
    const wrapper = shallow(<Card {...props} />);
    expect(wrapper.findWhere(n => n.hasClass("card")).length).toBe(1);
  });

  it("Link has path delivered from props", () => {
    const wrapper = shallow(<Card {...props} />);
    expect(wrapper.find("Link").prop("to")).toBe(props.link);
  });
});
