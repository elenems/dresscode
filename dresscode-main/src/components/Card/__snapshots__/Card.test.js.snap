// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Matches snapshot 1`] = `
"<div className=\\"card\\">
  <Link to=\\"link\\">
    <div className=\\"card__card-content\\">
      <Memo(Icon) name=\\"name\\" color=\\"black\\" />
      <div>
        <p />
        <span className=\\"card__link\\" />
      </div>
    </div>
  </Link>
</div>"
`;
